src/js/components/Tree/Tree.js

```diff
@@ -0,0 +1,161 @@
+                  {datum?.children?.length &&
+                    `${datum.children.length} sub-resources`}
+                </Text>
+                <FormNext />
+              </Box>
+            </Box>
+          );
+        }}
+      </List>
+    </Box>
+  );
+
+  const NestedNavigationList = () => (
+    <List
+      data={treeLevel}
+      onClickItem={event => {
+        setSelectedNode(event.item);
+      }}
+      {...theme.tree.nested.listProps}
+    >
+      {datum => {
+        return (
+          <Box direction="row" align="center" gap="small">
+            {selectedNode === datum ? <FormDown /> : <FormNext />}
+            <Text size="small">{datum.name}</Text>
+            {/* TODO
+            The concept of "nested" differs from  "column" 
+            by the following implementations aspects: 
+            1. selectedNode should be an array instead of a single value
+            2. focus should be per List item which isn't achievable via List
+            3. treeLevel and parentLevel are no longer relevant
+            4. concept of depth is no longer relevant as we may have 'multiple' 
+            5. should we consider leverage the existing mechanism of Collapsible
+            */}
+          </Box>
+        );
+      }}
+    </List>
+  );
+
+  let content;
+  if (mode === 'column') {
+    content = (
+      <Box align="start" gap="medium">
+        {/* Breadcrumbs */}
+        {depth > 0 && breadcrumbs.length > 0 && (
+          <Button
+            a11yTitle={breadcrumbs[breadcrumbs.length - 1]}
+            label={
+              <Text style={{ textDecoration: 'underline' }} size="small">
+                {breadcrumbs[breadcrumbs.length - 1]}
+              </Text>
+            }
+            icon={<FormPrevious />}
+            plain
+            onClick={() => {
+              // remove the last breadcrumb item when Previous is selected
+              breadcrumbs.pop();
+              setBreadcrumbs(breadcrumbs);
+              setDepth(depth - 1);
+              setSelectedNode(parentLevel);
+            }}
+          />
+        )}
+        {/* Layout Containers for List and user content */}
+        <Box direction="row" border width="large">
+          <ColumnNavigationList />
+          <Box border="left" align="start" gap="large" pad="medium">
+            {selectedNode && children(selectedNode)}
+            {selectedNode?.children?.length > 0 && (
+              <Button
+                a11yTitle={`${selectedNode.children.length} sub-resources`}
+                plain
+                label={
+                  <Text size="small">
+                    {`${selectedNode.children.length} sub-resources`}
+                  </Text>
+                }
+                icon={<FormNext />}
+                reverse
+                onClick={() => {
+                  // add a new selected entry to the breadcrumbs path
+                  setBreadcrumbs([...breadcrumbs, selectedNode.name]);
```

Comment:
I think we should allow something like `nameKey` or `labelKey` as a prop such that this would be `selectedNode[labelKey]`.